src/java_bytecode/java_utils.cpp

```diff
@@ -182,3 +184,16 @@ dereference_exprt checked_dereference(const exprt &expr, const typet &type)
   result.set(ID_java_member_access, true);
   return result;
 }
+
+void java_add_components_to_class(
+  symbolt &class_symbol,
+  const struct_union_typet::componentst &components_to_add)
+{
```

Comment:
Add `PRECONDITION`s on what is required of the class symbol (e.g. it is a type symbol, the type is a struct)